src/services/dynamo-db-service.ts

```diff
@@ -103,4 +89,25 @@ export class DynamoDbService {
     logger.debug(JSON.stringify(response));
     return response;
   }
+
+  public async updateDeleteStatus(userId: string) {
+    const ttl = getCurrentTimestamp().seconds + appConfig.maxRetentionSeconds;
+    const commandInput: UpdateItemCommandInput = {
+      TableName: this.tableName,
+      Key: { pk: { S: userId } },
+      UpdateExpression: 'SET #isAccountDeleted = :isAccountDeleted, #ttl = :ttl',
+      ExpressionAttributeNames: {
+        '#isAccountDeleted': 'isAccountDeleted',
+        '#ttl': 'ttl',
+      },
+      ExpressionAttributeValues: {
+        ':isAccountDeleted': { BOOL: true },
+        ':ttl': { N: ttl.toString() },
+      },
+      ConditionExpression: 'attribute_exists(pk)',
```

Comment:
lets add a condition that its only updated if `isAccountDeleted` is not present or is `false` otherwise the update fails but the message is ignored and redriven back to the queue